widget/list.go

```diff
@@ -257,6 +257,30 @@ func (l *List) ScrollToTop() {
 	l.Refresh()
 }
 
+// ScrollToOffset scrolls the list to the given offset position.
+//
+// Since: 2.5
+func (l *List) ScrollToOffset(offset float32) {
+	if l.scroller == nil {
+		return
+	}
+	if offset < 0 {
+		offset = 0
+	} else if h := l.contentMinSize().Height; offset > h {
```

Comment:
contentMinSize() is O(n) in the number of items with set heights. If we want to assume that using setItemHeight to make a height shorter than itemMin.Height is not allowed, we can have an O(1) check to skip calling contentMinSize if the offset is within the lower bound on content size of `numItems*itemMin.Height + separatorHeight*numSeparators`